saleor/tests/e2e/orders/test_order_products_on_fixed_promotion.py

```diff
@@ -0,0 +1,239 @@
+    permission_manage_discounts,
+    permission_manage_orders,
+):
+    # Before
+    channel_slug = "test-channel"
+    variant_price = "20"
+    promotion_name = "Promotion Fixed"
+    discount_value = 5
+    discount_type = "FIXED"
+    promotion_rule_name = "rule for product"
+
+    (
+        channel_id,
+        product_variant_id,
+        shipping_method_id,
+    ) = prepare_product(
+        e2e_staff_api_client,
+        permission_manage_products,
+        permission_manage_channels,
+        permission_manage_shipping,
+        permission_manage_product_types_and_attributes,
+        permission_manage_discounts,
+        permission_manage_orders,
+        channel_slug,
+        variant_price,
+        promotion_name,
+        discount_value,
+        discount_type,
+        promotion_rule_name,
+    )
+
+    # Step 1 - Create a draft order for a product with fixed promotion
+    lines = [{"variantId": product_variant_id, "quantity": 1}]
+    input = {
+        "channelId": channel_id,
+        "billingAddress": DEFAULT_ADDRESS,
+        "shippingAddress": DEFAULT_ADDRESS,
+        "shippingMethod": shipping_method_id,
+        "lines": lines,
+    }
+    data = draft_order_create(e2e_staff_api_client, input)
+    order_id = data["order"]["id"]
+    assert data["order"]["billingAddress"] is not None
+    assert data["order"]["shippingAddress"] is not None
+    order_shipping_method_id = data["order"]["shippingMethods"][0]["id"]
+    unit_price = float(variant_price) - float(discount_value)
+    undiscounted_price = data["order"]["lines"][0]["undiscountedUnitPrice"]["gross"][
+        "amount"
+    ]
+    assert float(undiscounted_price) == float(variant_price)
+    assert data["order"]["lines"][0]["unitPrice"]["gross"]["amount"] == unit_price
+    print(data)
+    # UNCOMMENT WHEN https://github.com/saleor/saleor/issues/13675 is resolved:
+    # assert order_shipping_method_id == shipping_method_id
+    assert order_id is not None
+
+    # Step 2 - Add order lines to the order
+    lines = [{"variantId": product_variant_id, "quantity": 1}]
+    order_lines = order_lines_create(e2e_staff_api_client, order_id, lines)
+    order_product_variant_id = order_lines["order"]["lines"][0]["variant"]["id"]
+    assert order_product_variant_id == product_variant_id
+    print(order_lines)
+
+    # Step 3 - Add a shipping method to the order
+    input = {"shippingMethod": shipping_method_id}
+    draft_update = draft_order_update(e2e_staff_api_client, order_id, input)
+    order_shipping_id = draft_update["order"]["deliveryMethod"]["id"]
+    shipping_price = draft_update["order"]["shippingPrice"]["gross"]["amount"]
+    subtotal_gross_amount = draft_update["order"]["subtotal"]["gross"]["amount"]
+    total_gross_amount = draft_update["order"]["total"]["gross"]["amount"]
+    assert order_shipping_method_id == order_shipping_id
+    print(draft_update)
+
+    # Step 4 - Complete the draft order
```

Comment:
Can you add an assertion that data.draftOrderComplete.order.lines[0].unitDiscountReason == Promotion rules discounts: {promotion_name}: {promotion_rule_name}

Reply:
https://github.com/saleor/saleor/pull/13639/commits/91a63d3c240131458d572a698fe6cac4b3077a58